src/session-pool.ts

```diff
@@ -681,25 +766,49 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
   /**
    * Borrows the first session from the inventory.
    *
+   * @param {boolean} longRunningTransaction Is transaction long-running.
    * @private
    *
    * @return {Session}
    */
-  _borrowFrom(): Session {
+  _borrowFrom(longRunningTransaction: boolean): Session {
     const session = this._inventory.sessions.pop()!;
     this._inventory.borrowed.add(session);
+    session.lastUsed = Date.now();
```

Comment:
Would it be better to call this property something like `checkedOutAt`. `lastUsed` sounds like the last time a statement was executed on this session.

Reply:
This value already existed and is also used by pinging mechanism to check is the session needs to be pinged